server/app.js

```diff
@@ -1,15 +1,38 @@
 const express = require('express');
+const handlebars = require('express-handlebars');
+const path = require('path');
+
 const someController = require('./controllers/some-controller');
 const anotherController = require('./controllers/another-controller');
 
+// James' controllers
+const getWorldCupResult = require('./controllers/get-world-cup-result-controller');
+const postWorldCupResult = require('./controllers/post-world-cup-result-controller');
+
 // Initializes a barebones express app
 const app = new express();
 
+// Set paths for views
+// https://www.npmjs.com/package/path
+app.set('views', path.join(__dirname, '/views'));
+
+// Sets up Handlebars for templating
+// https://github.com/ericf/express-handlebars
+app.engine('hbs', handlebars({
+		defaultLayout: 'main',
+		extname: '.hbs',
+		layoutsDir: path.join(__dirname, 'views/layouts/')
+	}
+));
+app.set('view engine', 'hbs');
```

Comment:
Good work!

There's an alternative to avoid the need to import the `path` library and use some handlebars defaults. Try it out to get a feel for it (if you haven't already). 

Neither is more correct than the other, but this second one might be easier to maintain (relies on handlebars defaults) and includes one less library.

```js
const handlebars = require('express-handlebars');
...
const handlebarsInstance = handlebars.create({extname: '.html'}); // alternatively .hbs
app.engine('html', handlebarsInstance.engine);
app.set('view engine', '.html'); // alternatively .hbs
```

This will set a few defaults automatically, including the `defaultLayout` to `main` inside `views/layouts`, and `partials` dir to `view/partials`. 

Read through the [handlebars docs](https://github.com/ericf/express-handlebars#extnamehandlebars) as you experiment with this to make sure it's clear what each bit is doing

Reply:
Ahhhh... that is awesome! I always dislike having to add more libraries than what is needed.

I was not sure about the instances when reading the documentation and whenever I would get it working. If I have this in the root folder, along with the views, controllers etc. Then it won't break, but as it's all in the server folder. I've looked to see how to set that as to look inside the server folder instead of looking for it in the root folder.